src/prefect/transactions.py

```diff
@@ -64,9 +65,18 @@ class Transaction(ContextModel):
     )
     overwrite: bool = False
     logger: Union[logging.Logger, logging.LoggerAdapter, None] = None
+    _hook_data: Dict[str, Any] = PrivateAttr(default_factory=dict)
```

Comment:
nit: I think it makes sense to name this attr more generally since the values can be used anywhere in a transaction, not just within hooks.
```suggestion
    _stored_values: Dict[str, Any] = PrivateAttr(default_factory=dict)
```